packages/EchoFramework/src/hooks/useHas3dAccess.ts

```
@@ -0,0 +1,35 @@
+import { BaseError, useInstCode } from '@equinor/echo-core';
+import { useEffect, useState } from 'react';
+import { check3dPermissionForPlantCode, check3dPermissionOnInstallation } from '../utils/plantInfo';
+/**
+ * Hook for checking if user has access to models for a given asset
+ *
+ * @export
+ * @param {string} [plantCode] asset to check access for, if not supplied instCode from user settings
+ * @param {((error: Error | BaseError | unknown) => void)} [handleError] method that will handle if an error occurred
+ * @return {*}  {boolean} returns wether or not the user have access
+ */
+export default function useHas3dAccess(
+    plantCode?: string,
+    handleError?: (error: Error | BaseError | unknown) => void
+): boolean {
+    const [has3dAccess, setHas3dAccess] = useState(true);
+    const instCode = useInstCode();
+
+    useEffect(() => {
+        async function checkAccess(): Promise<void> {
+            try {
+                const result = plantCode
+                    ? await check3dPermissionForPlantCode(plantCode)
+                    : await check3dPermissionOnInstallation(instCode);
+                setHas3dAccess(result);
+            } catch (error) {
+                if (handleError) handleError(error);
```

I don't think we can move things directly from echopedia, but we have to rewrite it: To make it easy to use from the outside, and also keep intact error reporting.

We could call handleError from core, but that requires analyticsModule as inArgument, which also makes this hook more difficult to use. Ideally this error handling should happen automatically behind the scene.

I see two options here:
1) Dependency injection / service locator. All analyticsModules are registered in core when they are created. And we fetch the correct analyticsModule by inspecting the route. This will be a general solution which we can re-use for all hooks for error reporting. 

2) Instead, rewrite this to be a normal function instead of a hook. Then we can throw the error as normal, and the outside is forced to deal with it. The hook itself has to be implemented in each module, and that is fairly easy since it's a small wrapper around the new function.

---

Sounds good, this is probably the best solution for now.

And thinking a bit more about it, 1 is probably wrong. Because we don't necessarily want to log all errors. Some errors can be expected in some cases, like not found, or forbidden. So it's better that we let the outside decide how to deal with the errors.